Sources/_Datadog_TestRunner/DDTestRunner.swift

```diff
@@ -0,0 +1,139 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+@_implementationOnly import XCTest
+
+internal class DDTestRunner: NSObject, XCTestObservation {
+    static var instance: DDTestRunner?
+
+    let testNameRegex = try? NSRegularExpression(pattern: "([\\w]+) ([\\w]+)", options: .caseInsensitive)
+    let supportsSkipping = NSClassFromString("XCTSkippedTestContext") != nil
+    var currentBundleName = ""
+    var activeTestSpan: DDSpan?
+
+    override init() {
+        super.init()
+        XCTestObservationCenter.shared.addTestObserver(self)
+        startTracer()
+    }
+
+    func startTracer() {
+        var clientToken = ""
+        if let envToken = ProcessInfo.processInfo.environment["DATADOG_CLIENT_TOKEN"] {
+            clientToken = envToken.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines)
+        } else if let bundleToken = Bundle.main.infoDictionary?["DatadogClientToken"] as? String {
+            clientToken = bundleToken.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines)
+        }
+
+        if clientToken.isEmpty {
+            return
+        }
+
+        Datadog.initialize(
+            appContext: .init(),
+            configuration: Datadog.Configuration
+                .builderUsing(clientToken: clientToken, environment: CIEnvironmentValues.getEnvVariable("DD_ENV") ?? "test")
+                .set(serviceName: CIEnvironmentValues.getEnvVariable("DD_SERVICE") ?? ProcessInfo.processInfo.processName)
+                .set(tracedHosts: ["."])
+                .build()
+        )
```

Comment:
Why exactly do we need to initialise the entire SDK here? Do we need all its features (Logging, Tracing and RUM) or maybe just the Tracing?

All the features inside the SDK are **fully isolated**. What happens in result of `Datadog.initialize()` is that we instantiate/or not each feature respectively (according to the configuration):
```swift
LoggingFeature.instance = LoggingFeature(...)
TracingFeature.instance = TracingFeature(...)
RUMFeature.instance = RUMFeature(...)
```

Maybe we don't need to call the entire `Datadog.initialize()` but just create the instance of `TracingFeature` to make the instrumentation `Tracer` work? If possible, that would be IMO recommended way, because `Datadog.initialize()` called here may bring some unwanted consequences (today or in the feature).

Reply:
The idea is collecting all possible information for the system under test for the duration of a test, being it a unit or an integration test. If the user has added logs or traces or some things are autoinstrumented, collecting them as part of the test would be very useful for understanding why the test has failed or why it is working slower. With this information, it is also very useful to write and run a test just to understand what the system is doing under the hood as a developer tool.  I have some doubts with RUM, don't still know if it could add value to a test, not unit test probably but maybe to integration tests.

Reply:
I see 👌. I'm only wondering, what can we do to avoid "double" initialization of the SDK? Let's say the customer app calls `Datadog.initialize()` in his `AppDelegate`, which is also invoked in unit tests / UI Tests. Also, what if the user initialization leads to specific SDK state, i.e. `.enableTracing(false)`?

Reply:
We should not let the SDK to be initialised twice, we should fail if already initialised in tests. Also the configuration for the tests is probaby not the same that the user may desire for production, we cannot let the user disable tracing while testing for example (it should disable al the test instrumentation instead). We can start by doing all configuration to the test tracer by env variables, and see in the future how can the user modify it in code if necessary.

Reply:
Sounds good 👍. We're currently failing (with a console warning) on doubled initialisation, but we could do it silently in tests 👌.